lib/auth/trustedcluster.go

```diff
@@ -180,6 +199,10 @@ func (a *AuthServer) establishTrust(trustedCluster services.TrustedCluster) erro
 	// send the request to the remote auth server via the proxy
 	validateResponse, err := a.sendValidateRequestToProxy(trustedCluster.GetProxyAddress(), &validateRequest)
 	if err != nil {
+		log.Error(err)
+		if strings.Contains(err.Error(), "x509") {
```

Comment:
I think you can use type casting here that is more robust than string submatch:

```
case x509.SystemRootsError, x509.UnknownAuthorityError:
```

Reply:
I thought so too. Unfortunately that's not an error from the http package, it's our own JSON error from the API.

Reply:
`trace.IsTrustError()` fails on it too.